tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,500 @@
+        paging_info: The paging information used in query parameters.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+
+    query_string = {}
+    if group_id is not None:
+        query_string["groupId"] = group_id
+    if search is not None:
+        query_string["query"] = search
+    if paging_info is not None:
+        query_string["index"] = paging_info["index"]
+        query_string["pageLength"] = paging_info["page_length"]
+
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}",
+        query_string=query_string,
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["data"] == expected
+
+
+def assert_registering_existing_queue_name_fails(
+    client: FlaskClient, name: str, group_id: int
+) -> None:
+    """Assert that registering a queue with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        name: The name to assign to the new queue.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = actions.register_queue(client, name=name, group_id=group_id)
+    assert response.status_code == 400
+
+
+def assert_queue_name_matches_expected_name(
+    client: FlaskClient, queue_id: int, expected_name: str
+) -> None:
+    """Assert that the name of a queue matches the expected name.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+        expected_name: The expected name of the queue.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the name of the
+            queue does not match the expected name.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_queue_is_not_found(
+    client: FlaskClient,
+    queue_id: int,
+) -> None:
+    """Assert that a queue is not found.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+
+    Raises:
+        AssertionError: If the response status code is not 404.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 404
+
+
+def assert_cannot_rename_queue_with_existing_name(
+    client: FlaskClient,
+    queue_id: int,
+    existing_name: str,
+    existing_description: str,
+) -> None:
+    """Assert that renaming a queue with an existing name fails.
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        name: The name of an existing queue.
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
```

Comment:
Whitespace nitpick for the docstring, although I think the Google-style docstring style requires separating the sections with a newline:

```suggestion
    """Assert that renaming a queue with an existing name fails.

    Args:
        client: The Flask test client.
        queue_id: The id of the queue to rename.
        name: The name of an existing queue.

    Raises:
        AssertionError: If the response status code is not 400.
    """
```